### ISSUE TITLE
 Implement a Custom Trainer with Adjustable Batch Size and Learning Rate

### ISSUE THREAD

**Issue by Antlera** [intelligent-machine-learning/dlrover#549]
The goal of this issue is to design and implement a `custom trainer` that can dynamically adjust hyperparameters, notably the `batch size` and `learning rate`, during the training process. 
This trainer should have the ability to pause the ongoing training, apply new hyperparameters.
The feature will allow us to use our GPU resources more efficiently.

**Comment by Antlera** [comment]
An `elastic dataloader is needed in this component.

**Comment by Antlera** [comment]
Closed as completed in DLRover autotuner.
